src/ghostwriter/models/v1/mapping.py

```diff
@@ -0,0 +1,180 @@
+"""Definition of the mapping from a top-level route to its substituted form,
```

Comment:
I'm wondering if it's worth moving the business logic out, perhaps into services/ & hooks and keep this to just the model definitions?
Something like:

- services/
    - route_resolver.py (resolve, run_hooks)
- hooks /
    - hook_definitions.py

Reply:
I want to keep the hooks standalone.  And the resolution...well, maybe it's a service?  But I kind of feel like the model shouldn't depend on much of anything (this is the only versioned external model anyway, because this definition is the thing you really have to write for each site), and the only thing this model is even _for_ is to rewrite the route.  So it kinda felt like a property of the model.

Reply:
Keeping it like it is.